### PR TITLE
perf(js): Remove jquery-migrate and upgrade jquery to 2.0

### PR DESCRIPTION
**Pull request by ewinslow**
BREAKING CHANGE:
We've removed jquery-migrate from core and upgraded jquery to ^2.1.4

See http://jquery.com/upgrade-guide/1.9/ for how to move off jquery-migrate.

If you'd prefer to just add it back, you can use this code in your plugin's init:

``` php
elgg_register_js('jquery-migrate', elgg_get_simplecache_url('jquery-migrate.js'), 'head');
elgg_load_js('jquery-migrate');
```

Also, define a `jquery-migrate.js` view containing the contents of the script.

### CODE REVIEW

**Comment by mrclay** [comment]
Feels too late for 2.0. I know there's core code using `live()` and probably more. I think until we update our core code we shouldn't push this on plugin devs.

We've already made it super easy to remove the migrate.

File: js/lib/ui.js [hunk 1]
```diff
@@ -9,20 +9,20 @@ elgg.ui.init = function () {
 	elgg.ui.initHoverMenu();
 
 	//if the user clicks a system message, make it disappear
-	$('.elgg-system-messages li').live('click', function() {
+	$(document).on('click', '.elgg-system-messages li', function() {
```
Comment (mrclay) [inline js/lib/ui.js]: Oh I didn't see you fixed these

File: mod/thewire/views/default/js/thewire.php [hunk 2]
```diff
@@ -14,12 +14,10 @@
 		}
 	};
 
-	$("#thewire-textarea").live({
-		input: callback,
-		onpropertychange: callback
-	});
+	$(document).on('input', "#thewire-textarea", callback);
+	$(document).on('onpropertychange', '#thewire-textarea', callback);
```
Comment (mrclay) [inline mod/thewire/views/default/js/thewire.php]: it's not just "propertychange"?

Comment (ewinslow) [inline mod/thewire/views/default/js/thewire.php]: Thought that was strange too, buy just keeping it consistent with his the object was structured... Can add a separate fix if needed.

**Comment by ewinslow** [comment]
Given that all core used of live are removed, how do you feel about this change?

**Comment by mrclay** [comment]
Definitely for the jQuery upgrade.

**Comment by ewinslow** [comment]
OK I will pull this in after the js/ css/ prefix change goes in unless I hear otherwise.

File: mod/thewire/views/default/js/thewire.php (hunk 2 above)
Comment (ewinslow) [inline mod/thewire/views/default/js/thewire.php]: Yea, this should just be propertychange. I could probably even register both of them in one go with:

```
$(document).on('input propertychange', '#thewire-textarea', callback);
```

**Pull request merged**